pymatgen/entries/corrections_calc.py

```diff
@@ -0,0 +1,504 @@
+"""
+This module calculates corrections for the species listed below, fitted to the experimental and computed
+entries given to the CorrectionCalculator constructor.
+"""
+# pylint: disable=C0330
+
+import warnings as w
+from collections import OrderedDict
+from typing import Dict, List, Tuple
+
+import numpy as np
+import plotly.graph_objects as go
+import ruamel.yaml
+from monty.serialization import loadfn
+from scipy.optimize import curve_fit
+
+from pymatgen import Composition, Element
+from pymatgen.analysis.reaction_calculator import ComputedReaction
+from pymatgen.analysis.structure_analyzer import sulfide_type
+
+
+def _func(x, *m):
+    """
+    Helper function for curve_fit.
+    """
+    return np.dot(x, m)
+
+
+class CorrectionCalculator:
+
+    """
+    A CorrectionCalculator contains experimental and computed entries which it uses to compute corrections.
+
+    It graphs residual errors after applying the computed corrections and creates the MPCompatibility.yaml
+    file the Correction classes use.
+
+    Attributes:
+        species: list of species that corrections are being calculated for
+        exp_compounds: list of dictionaries which each contain a compound's formula and experimental data
+        calc_compounds: dictionary of ComputedEntry objects
+        corrections: list of corrections in same order as species list
+        corrections_std_error: list of the variances of the corrections in same order as species list
+        corrections_dict: dictionary of format {'species': (value, uncertainty)} for easier correction lookup
+    """
+
+    species = [
```

Comment:
Since we went through a selection process to determine which species to calculate corrections for, I expected most common use case would be to calculate corrections for that same set of species and hard-coded it in. Do we want to give the user the option to specify the species themselves, perhaps upon initialization?

Reply:
Yes @awvio I think that would be good!